src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -73,201 +76,132 @@ public RoslynAnalyzerProvider(IAnalyzerInstaller analyzerInstaller, ILogger logg
             {
                 throw new ArgumentNullException(nameof(inactiveRules));
             }
+
             if (activeRules == null)
             {
                 throw new ArgumentNullException(nameof(activeRules));
             }
 
-            var analyzer = ConfigureAnalyzer(language, activeRules, inactiveRules);
-            if (analyzer == null)
-            {
-                this.logger.LogInfo(Resources.RAP_NoPluginInstalled, language);
-            }
-
-            return analyzer;
-        }
+            var rulesetFilePath = CreateRuleSet(language, activeRules, inactiveRules);
+            var analyzersAssemblies = FetchAnalyzerAssemblies(language, activeRules);
+            var additionalFiles = WriteAdditionalFiles(language, activeRules);
 
-        public static string GetRoslynFormatName(string language)
-        {
-            return string.Format(RoslynFormatNamePrefix, language);
+            return new AnalyzerSettings(language, rulesetFilePath, analyzersAssemblies, additionalFiles);
         }
 
         public static string GetRoslynRulesetFileName(string language)
         {
             return string.Format(RoslynRulesetFileName, language);
         }
 
-        #endregion Public methods
+        private string CreateRuleSet(string language, IEnumerable<SonarRule> activeRules, IEnumerable<SonarRule> inactiveRules)
+        {
+            var ruleSetGenerator = new RoslynRuleSetGenerator(this.serverProperties);
 
-        #region Private methods
+            var ruleSet = ruleSetGenerator.Generate(language, activeRules, inactiveRules);
 
-        /// <summary>
-        /// Generates several files related to rulesets and roslyn analyzer assemblies.
-        /// Active rules should never be empty, but depending on the server settings of repo keys, we might have no rules in the ruleset.
-        /// In that case, this method returns null.
-        /// </summary>
-        private AnalyzerSettings ConfigureAnalyzer(string language, IEnumerable<SonarRule> activeRules, IEnumerable<SonarRule> inactiveRules)
-        {
-            var ruleSetGenerator = new RoslynRuleSetGenerator(this.sqServerSettings);
-            var ruleSet = ruleSetGenerator.Generate(activeRules, inactiveRules, language);
-            var rulesetFilePath = WriteRuleset(ruleSet, language);
-            if (rulesetFilePath == null)
-            {
-                // no ruleset, nothing was written in disk
-                return null;
-            }
+            Debug.Assert(ruleSet != null, "Expecting the RuleSet to be created.");
+            Debug.Assert(ruleSet.Rules != null, "Expecting the RuleSet.Rules to be initialized.");
```

Comment:
These two used to be hard checks with log (see WriteRuleSet on the left, deleted), but since the Generate method will never return null, I converted to Debug.Assert.